src/node/jsonrpc.rs

```diff
@@ -0,0 +1,149 @@
+use crate::node::config::{
+    IPCAgentJsonRPCNodeConfig, DEFAULT_JSON_RPC_ENDPOINT, DEFAULT_JSON_RPC_VERSION,
+};
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use std::convert::Infallible;
+use warp::Filter;
+
+/// The IPC JSON RPC node that contains all the methods and handlers. The underlying implementation
+/// is using `warp`.
+///
+/// Note that currently only http json rpc is supported.
+///
+/// # Examples
+/// ```no_run
+/// use agent::node::config::IPCAgentJsonRPCNodeConfig;
+/// use agent::node::jsonrpc::IPCAgentJsonRPCNode;
+///
+/// #[tokio::main]
+/// async fn main() {
+///     let n = IPCAgentJsonRPCNode::new(IPCAgentJsonRPCNodeConfig::default());
```

Comment:
`addr` is an `Option` in the config, are we sure that this works? If I am not mistaken the default for an `Option` is `None`. I don't know the behavior of `run` over a `None` address, but we should probably figure out a better way to pass a default configuration.

Reply:
in the code, it's calling `config.addr()` which handles the `None` and convert to the address to a predefined socket addr.

Reply:
The other way I considered was actually in serde default deserialization, but that would result in losing the `#[derive(Default)]` and one cannot simply do:
```
IPCAgentJsonRPCNodeConfig{ addr: ..., ..Default::default() }
```
Any other methods you have in mind?

Reply:
Oh, I see... so `addr` is private and when accessing it through `addr()` it auto-assigns the default value. The thing that I am now wondering, is if once we force the use of a config file as input we need `addr` to be an `Option` at all, but I guess we can discuss that in the next iteration.

Reply:
Yep, make sense. If we are using the config file, I dont even need this config struct. I can use that config file. But just need to define a field for this addr.